scilpy/tractanalysis/scoring.py

```diff
@@ -0,0 +1,389 @@
+    """
+
+    mask_1_img = nib.load(mask_1_filename)
+    mask_2_img = nib.load(mask_2_filename)
+    mask_1 = get_data_as_mask(mask_1_img)
+    mask_2 = get_data_as_mask(mask_2_img)
+
+    if dilate_endpoints:
+        mask_1 = binary_dilation(mask_1, iterations=dilate_endpoints)
+        mask_2 = binary_dilation(mask_2, iterations=dilate_endpoints)
+
+    # TODO: Handle streamline IDs instead of streamlines
+    tmp_sft, sft = extract_streamlines(mask_1, mask_2, sft)
+
+    streamlines = tmp_sft.streamlines
+    tc_streamlines = streamlines
+    wpc_streamlines = []
+    fc_streamlines = []
+    nc_streamlines = []
+
+    # Config file for each 'bundle'
+    # Loops => no connection (nc) # TODO Is this legit ?
+    # Length => false connection (fc) # TODO Is this legit ?
+    if gt_config:
+        min_len, max_len = \
+            length_dict[gt_bundle]['length']
+
+        # Bring streamlines to world coordinates so proper length
+        # is calculated
+        tmp_sft.to_rasmm()
+        streamlines = tmp_sft.streamlines
+        lengths = np.array(list(length(streamlines)))
+        tmp_sft.to_vox()
+        streamlines = tmp_sft.streamlines
+
+        valid_min_length_mask = lengths > min_len
+        valid_max_length_mask = lengths < max_len
+        valid_length_mask = np.logical_and(valid_min_length_mask,
+                                           valid_max_length_mask)
+        streamlines = ArraySequence(streamlines)
+
+        val_len_streamlines = streamlines[valid_length_mask]
+        fc_streamlines = streamlines[~valid_length_mask]
+
+        angle = length_dict[gt_bundle]['angle']
+        tc_streamlines_ids = remove_loops_and_sharp_turns(
+            val_len_streamlines, angle)
+
+        loop_ids = np.setdiff1d(
+            range(len(val_len_streamlines)), tc_streamlines_ids)
+
+        loops = val_len_streamlines[list(loop_ids)]
+        tc_streamlines = val_len_streamlines[list(tc_streamlines_ids)]
+
+        if loops:
+            nc_streamlines = loops
+
+    # Streamlines getting out of the bundle mask can be considered
+    # separately as wrong path connection (wpc)
+    # TODO: Maybe only consider if they cross another GT bundle ?
+    if wrong_path_as_separate:
+        tmp_sft = StatefulTractogram.from_sft(tc_streamlines, sft)
+        _, wp_ids = filter_grid_roi(
+            tmp_sft, gt_bundle_inv_mask, 'any', False)
+        wpc_streamlines = tmp_sft.streamlines[list(wp_ids)]
+        tc_ids = np.setdiff1d(range(len(tmp_sft)), wp_ids)
+        tc_streamlines = tmp_sft.streamlines[list(tc_ids)]
+
+    tc_sft = StatefulTractogram.from_sft(tc_streamlines, sft)
+    wpc_sft = StatefulTractogram.from_sft([], sft)
+    fc_sft = StatefulTractogram.from_sft(fc_streamlines, sft)
+    if wrong_path_as_separate and len(wpc_streamlines):
+        wpc_sft = StatefulTractogram.from_sft(wpc_streamlines, sft)
+
+    return tc_sft, wpc_sft, fc_sft, nc_streamlines, sft
+
+
+def extract_false_connections(
```

Comment:
Dont know if its a coding standard, but we often put as much parameters on the line, then skip and repeat